### PR TITLE
Rails backend

### PR DESCRIPTION
**Pull request by haliljon**
## In this PR I was able to:

> - Implement a connection between a Ruby on Rails back-end and React front-end.
> - Understand the pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.

Here is the [Front-end](https://github.com/haliljon/hello-react-front-end.git) of the app.

When running Backend can you please make sure that you use 'http://localhost:3000' for the backend?

### CODE REVIEW

File: app/controllers/api/v1/greeting_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::GreetingController < ApplicationController
+  def index
+    @greetings = Greeting.all
```
Comment (anagudelogu) [inline app/controllers/api/v1/greeting_controller.rb]: - [ ] You did a great job working on this milestone! 💯 However, according to the requirements:

image

This endpoint should return one random Greeting and not all greetings. Please consider fixing this! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @haliljon, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of GitHub flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

- [ ] I'm getting this error when I try to use both apps together:

image

This is because you haven't set any CORS middleware in your server and it is blocking all external requests (including the one coming from your React App). You can use the [`rack-cors`](https://github.com/cyu/rack-cors) gem to fix this! 🤓 🔧 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/greeting_controller.rb (hunk 1 above)
Comment (haliljon) [reply app/controllers/api/v1/greeting_controller.rb]: Hi @anagudelogu 

It is done ✅

**Review by mwanawabangona — APPROVED** [review]
Hi @haliljon 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**